packages/app-config-writer/src/smartlinks-config/generateSmartLinks.ts

```diff
@@ -0,0 +1,49 @@
+import { create as createStorage } from 'mem-fs';
+import { create } from 'mem-fs-editor';
+import type { Editor } from 'mem-fs-editor';
+import type { ToolsLogger } from '@sap-ux/logger';
+import type { SmartLinksConfig } from '..';
+import { getCredentials, getServices, writeSmartLinksConfig } from '.';
+
+/**
+ * @description Add smartlinks configuration to a UI5 application.
+ * @param basePath - the base path where the ui5-deploy/ui5.yaml is
+ * @param config - configuration of the smartlinks
+ * @param logger - logger
+ * @param fs - the memfs editor instance
+ * @returns Promise<Editor> - memfs editor instance with updated files
+ */
+export async function generateSmartLinksConfig(
+    basePath: string,
+    config: SmartLinksConfig,
+    logger?: ToolsLogger,
+    fs?: Editor
+): Promise<Editor> {
+    if (!fs) {
+        fs = create(createStorage());
+    }
+    if (!config.service) {
+        const service = (await getServices(basePath, logger, fs))[0];
```

Comment:
It is not good idea just to take the first service. If there are multiple defined, then we need to ask the user to pick one.

Reply:
OK, will rework this

Reply:
Approach reworked and as discussed, if no `ui5-deploy.yaml` is found, user input is expected. No fallback to `ui5.yaml` for now.